### PR TITLE
sp/1.9.0-update-automatic-machine-learning-with-test-drive

### PR DESCRIPTION
**Pull request by 5675sp**

### CODE REVIEW

**Comment by james94** [comment]
@5675sp @Castro38 @FrankJVA  the opening sentence of paragraph 2 in **Data Preparation** in the concepts section can be revised. It is currently:

- "Data wrangling, which can include extractions, parsing, joining, standardizing, augmenting, cleansing, consolidating, missing data is fixed or removed. "

One suggestion for revision: 

- "Data wrangling can include extractions, parsing, joining, standardizing, augmenting, cleansing, and consolidating until the missing data is fixed or removed. "

**Comment by james94** [comment]
@5675sp the last sentence in **Algorithm Selection** in the concepts section has a duplicate. Random forest(s) is stated twice. It is currently:

- "Some of those algorithms include linear regression, regression trees, random forests, naive Bayes, and random forest, boosting, to name a few [5]."

One suggestion for revision is to remove the duplicate:

- "Some of those algorithms include linear regression, regression trees, random forests, naive Bayes, and boosting, to name a few [5]."

**Comment by james94** [comment]
@5675sp the third paragraph in **Model Training** located below the diagram has some grammatical errors. It is currently:

- "Another part of model training is fitting and tuning the models. For fitting and tuning, hyperparameters need to be tuned, and cross-validation needs to take place using only the training data. Various hyperparameters values will need to be tested. "A hyperparameter is a parameter that is set before the learning process begins. These parameters are tunable and can directly affect how well a model trains. An example of hyper parameter in machine learning is learning rate" [7]. With cross validation, the whole dataset is utilized, and each model is trained on a different subset of the training data [8]. Additionally, a cross-validation loop will be set to calculate the cross-validation score for each set of hyperparameters for each algorithm. Based on the cross-validation score and hyperparameter values, you can select the model(remember that "a model in ML is the output of a machine learning algorithm run on data. It represents whatt was learned by a machine learning algorithm" [9]) for each algorithm that has been tuned with training data and test it using your test set."

One suggestion for revision:

- "Another part of model training is fitting and tuning the models. For fitting and tuning, hyperparameters need to be tuned, and cross-validation needs to take place using only the training data. Various **`hyperparameter`** values will need to be tested. "A hyperparameter is a parameter that is set before the learning process begins. These parameters are tunable and can directly affect how well a model trains. An example of **`hyperparameter`** in machine learning is **`the`** learning rate" [7]. With **`cross-validation`**, the whole dataset is utilized, and each model is trained on a different subset of the training data [8]. Additionally, a cross-validation loop will be set to calculate the cross-validation score for each set of hyperparameters for each algorithm. Based on the cross-validation score and hyperparameter values, you can select the model(remember that "a model in ML is the output of a machine learning algorithm run on data. It represents **`what`** was learned by a machine learning algorithm" [9]) for each algorithm that has been tuned with training data and **`tested with`** your test set."

**Comment by 5675sp** [comment]
> @5675sp @Castro38 @FrankJVA the opening sentence of paragraph 2 in **Data Preparation** in the concepts section can be revised. It is currently:
> 
> * "Data wrangling, which can include extractions, parsing, joining, standardizing, augmenting, cleansing, consolidating, missing data is fixed or removed. "
> 
> One suggestion for revision:
> 
> * "Data wrangling can include extractions, parsing, joining, standardizing, augmenting, cleansing, and consolidating until the missing data is fixed or removed. "

Just fixed it, thanks for the feedback.

**Comment by 5675sp** [comment]
> @5675sp the last sentence in **Algorithm Selection** in the concepts section has a duplicate. Random forest(s) is stated twice. It is currently:
> 
> * "Some of those algorithms include linear regression, regression trees, random forests, naive Bayes, and random forest, boosting, to name a few [5]."
> 
> One suggestion for revision is to remove the duplicate:
> 
> * "Some of those algorithms include linear regression, regression trees, random forests, naive Bayes, and boosting, to name a few [5]."

fixed, thanks

**Comment by 5675sp** [comment]
> @5675sp the third paragraph in **Model Training** located below the diagram has some grammatical errors. It is currently:
> 
> * "Another part of model training is fitting and tuning the models. For fitting and tuning, hyperparameters need to be tuned, and cross-validation needs to take place using only the training data. Various hyperparameters values will need to be tested. "A hyperparameter is a parameter that is set before the learning process begins. These parameters are tunable and can directly affect how well a model trains. An example of hyper parameter in machine learning is learning rate" [7]. With cross validation, the whole dataset is utilized, and each model is trained on a different subset of the training data [8]. Additionally, a cross-validation loop will be set to calculate the cross-validation score for each set of hyperparameters for each algorithm. Based on the cross-validation score and hyperparameter values, you can select the model(remember that "a model in ML is the output of a machine learning algorithm run on data. It represents whatt was learned by a machine learning algorithm" [9]) for each algorithm that has been tuned with training data and test it using your test set."
> 
> One suggestion for revision:
> 
> * "Another part of model training is fitting and tuning the models. For fitting and tuning, hyperparameters need to be tuned, and cross-validation needs to take place using only the training data. Various **`hyperparameter`** values will need to be tested. "A hyperparameter is a parameter that is set before the learning process begins. These parameters are tunable and can directly affect how well a model trains. An example of **`hyperparameter`** in machine learning is **`the`** learning rate" [7]. With **`cross-validation`**, the whole dataset is utilized, and each model is trained on a different subset of the training data [8]. Additionally, a cross-validation loop will be set to calculate the cross-validation score for each set of hyperparameters for each algorithm. Based on the cross-validation score and hyperparameter values, you can select the model(remember that "a model in ML is the output of a machine learning algorithm run on data. It represents **`what`** was learned by a machine learning algorithm" [9]) for each algorithm that has been tuned with training data and **`tested with`** your test set."

fixed, thanks

**Comment by james94** [comment]
@5675sp thank you for making the updates.

**Comment by Castro38** [comment]
Thank you @5675sp!
